src/masternodes/vaulthistory.h

```diff
@@ -55,14 +55,62 @@ struct VaultHistoryValue {
     }
 };
 
+struct VaultStateKey {
+    uint256 vaultID;
+    uint32_t blockHeight;
+
+    ADD_SERIALIZE_METHODS;
+
+    template <typename Stream, typename Operation>
+    inline void SerializationOp(Stream& s, Operation ser_action) {
+        READWRITE(vaultID);
+
+        if (ser_action.ForRead()) {
+            READWRITE(WrapBigEndian(blockHeight));
+            blockHeight = ~blockHeight;
+        }
+        else {
+            uint32_t blockHeight_ = ~blockHeight;
+            READWRITE(WrapBigEndian(blockHeight_));
+        }
+    }
+};
+
+struct VaultStateValue {
+    bool isUnderLiquidation;
```

Comment:
`auctionBatches` not empty means in liquidation.

Reply:
Updated.